### PR TITLE
Make classification cancellable.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by vladima** [comment]
:+1:

**Comment by DanielRosenwasser** [comment]
:+1:

File: src/services/services.ts [hunk 1]
```diff
@@ -5983,6 +5983,26 @@ namespace ts {
...
 
+        function checkForClassificationCancellation(kind: SyntaxKind) {
+            // We don't want to actually call back into our host on every node to find out if we've
+            // been canceled.  That would be an enormous amount of chattyness, along with the all
+            // the overhead of marshalling the data to/from the host.  So instead we pick a few
+            // reasonable node kinds to bother checking on.  These node kinds represent high level
+            // constructs that we would expect to see commonly, but just at a far less frequent 
+            // interval.
+            //
+            // For example, in checker.ts (around 750k) we only have around 600 of these constructs.
```
Comment (mhegazy) [inline src/services/services.ts]: We could set an interval instead and query if the interval has not elapsed since last query.

Comment (CyrusNajmabadi) [inline src/services/services.ts]: Will do!

**Pull request merged**